src/audio/copier/copier_gain.h

```diff
@@ -0,0 +1,144 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2024 Intel Corporation.
+ *
+ * Author: Ievgen Ganakov <ievgen.ganakov@intel.com>
+ */
+
+#ifndef __SOF_COPIER_GAIN_H__
+#define __SOF_COPIER_GAIN_H__
+
+#include <sof/audio/buffer.h>
+#include <ipc4/base_fw.h>
+#include <ipc/dai.h>
+#if SOF_USE_HIFI(3, COPIER) || SOF_USE_HIFI(4, COPIER) || SOF_USE_HIFI(5, COPIER)
+#include <xtensa/tie/xt_hifi3.h>
+#endif
+/**
+ * @file copier_gain.h
+ * @brief Header file containing definitions and functions related to audio gain
+ * processing for a copier module.
+ *
+ * This file provides functions, constants and structure definitions for applying gain to
+ * input audio buffers, both in 16-bit and 32-bit container formats. The gain can be
+ * applied in different directions (addition or subtraction) and has three modes:
+ * - static gain
+ * - transition gain (fade-in/fade-out)
+ * - mute
+ */
+
+/* Maximum number of gain coefficients */
+#define MAX_GAIN_COEFFS_CNT 4
+
+/* Common const values for applying gain feature */
+#define Q10_TO_Q31_SHIFT 6
+#define Q10_TO_Q15_SHIFT 5
+
+/* 16x2 store operation requires shift to middle part of 32 bit register */
+#define I64_TO_I16_SHIFT  48
+#define MIDDLE_PART_SHIFT 8
+
+/* Unit gain in q10 format applied by default */
+#define UNITY_GAIN_4X_Q10  0x0400040004000400ULL
+#define UNITY_GAIN_GENERIC 0x0400U
+#define MAX_INT64          0x7FFFFFFFFFFFFFFFULL
```

Comment:
Why unsigned for generic and max? E.g.  MAX_INT64 fits signed type and I think the code, especially HiFi version is using signed types. Add also  somewhere comment what is the max range of gain in linear and decibels.

Reply:
Fixed